src/data.rs

```diff
@@ -1,142 +1,152 @@
-            .chunks(group_length)
-            .map(|chunk| chunk.to_vec())
-            .collect()
-    }
+/**
+ * Class to contain the data processing functions
+ */
+pub struct ProcessData {}
 
+impl ProcessData {
+    /**
+     * Computes the twos complement of the given value.
+     */
     pub fn twos_comp(val: u32, bits: usize) -> i64 {
-        // Computes the twos complement of the given value.
         if (val & (1 << (bits - 1))) != 0 {
-            (val as i64) - (1 << bits)
+            (val as i64) - ((1 << bits) as i64)
         } else {
             val as i64
         }
     }
 
+    /**
+     * Transforms the given data bytes into a value in little endian.
+     * Little Endian byte order stores low order bytes first.
+     */
     pub fn little_endian(data_bytes: &[u8], bits: usize) -> u32 {
-        // Transforms the given data bytes into a value in little endian.
-        // Little Endian byte order stores low order bytes first.
         let mut result: u32 = 0;
         for (i, byte) in data_bytes.iter().enumerate() {
-            // println!("Little End Byte: {}", byte);
             result |= (*byte as u32) << (bits * i);
-            // println!("Little End Result: {}", result)
         }
         result
     }
 
+    /**
+     * Transforms the given data bytes into a value in big endian.
+     * Big Endian byte order stores low order bytes last.
+     */
     pub fn big_endian(bytes: &[u8], bits: usize) -> u32 {
-        // Transforms the given data bytes into a value in big endian.
-        // Big Endian byte order stores low order bytes last.
         let mut result: u32 = 0;
         for (i, byte) in bytes.iter().enumerate() {
-            // println!("Big End Byte: {}", byte);
             result |= (*byte as u32) << (bits * (bytes.len() - i - 1));
-            // println!("Big End Result: {}", result);
         }
         result
     }
 
-    pub fn default_decode(byte_vals: &[u8]) -> Vec<i64> {
-        // Default decode structure seen by a majority of the messages.
-
-        let grouped_vals = ProcessData::group_bytes(byte_vals, 2);
-        println!("CUCKED GROUP BYTES");
-        let parsed_vals: Vec<u32> = grouped_vals
-            .iter()
-            .map(|val| ProcessData::little_endian(val, 8))
-            .collect();
-        println!("CUCKED LITTLE ENDIAN");
-        let decoded_vals: Vec<i64> = parsed_vals
-            .iter()
-            .map(|val| ProcessData::twos_comp(*val, 16))
-            .collect();
-        println!("CUCKED TWOS COMP");
-        decoded_vals
-    }
+    /**
+     * Decodes the given byte by taking the top four bits after shifting it by the given number of bits.
+     */
+    pub fn half(byte: u8, bits: u8) -> u32 {
+        (byte >> bits & 15) as u32
+    } 
 }
 
-pub struct FormatData {
-    // Utility functions to scale data values of a specific type.
-}
+/**
+ * Class to contain the data formatting functions
+ */
+pub struct FormatData {}
 
 impl FormatData {
```

Comment:
I think that these FormatData functions are a bit ambiguous and use a lot of magic numbers. We should probably add either a comment explaining where the normalization factor is coming from and what type of data should be used with it

Reply:
I agree the issue is that most of these are legacy from peter so i dont have the best idea on what the majority are